File: mater-rest/src/main/java/com/oneune/mater/rest/main/store/dtos/SaleLinkDto.java
package com.oneune.mater.rest.main.store.dtos;

import com.oneune.mater.rest.main.store.enums.SaleStatusEnum;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import java.time.Instant;

/**
 * @see com.oneune.mater.rest.main.store.entities.SaleLinkEntity
 */
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Data
public class SaleLinkDto extends AbstractDto {
    UserDto buyer;
    CarDto car;
    Instant createdAt;
    SaleStatusEnum status;
    Float score;
}

File: mater-webapp/src/app/store/dtos/sale-link.dto.ts
import {UserDto} from "./user.dto";
import {CarDto} from "./car.dto";
import {SaleLinkStatusEnum} from "../enums/sale-link-status.enum";
import {AbstractDto} from "./core/abstract.dto";

export class SaleLinkDto extends AbstractDto {
  buyer: UserDto;
  car: CarDto;
  createdAt: Date;
  status: SaleLinkStatusEnum;
  score: number;
  note: string;
}
